filament/src/fg/FrameGraph.cpp

```diff
@@ -284,12 +285,11 @@ FrameGraph& FrameGraph::compile() noexcept {
             resourceNodes[resource.index]->readerCount++;
         }
 
-#ifndef NDEBUG
+        // set the writers
         for (FrameGraphHandle resource : pass.writes) {
             assert(resource.isValid());
-            assert(resourceNodes[resource.index]->writer == &pass);
+            resourceNodes[resource.index]->writer = &pass;
```

Comment:
Is it normal that the test becomes an assignment?

Reply:
yes -- it used to be this way (which is the correct way). But at one point I moved the assignment somewhere else (where it's incorrect), and I added the assert() here to make sure they matched.